### PR TITLE
CPR-440 Add currently managed to person record

### PR DESCRIPTION
**Pull request by H-Iwanejko**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/court/commonplatform/Defendant.kt [hunk 1]
```diff
@@ -4,10 +4,10 @@ import com.fasterxml.jackson.annotation.JsonIgnoreProperties
 import com.fasterxml.jackson.annotation.JsonProperty
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize
 import jakarta.validation.Valid
+import uk.gov.justice.digital.hmpps.personrecord.client.model.prisoner.deserializers.CROIdentifierDeserializer
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/court/commonplatform/Defendant.kt:7]: not sure about these being in a `prisoner` package - CROs are not only present on prisoners, same goes for PNCs

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/prisoner/CurrentlyManaged.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      INACTIVE_TRANSFER("Inactive: TRN"),
+      INACTIVE_OUT("Inactive: OUT"),
+    }
+
+    fun from(inputStatus: String? = EMPTY_STATUS): CurrentlyManaged {
+      val status: Boolean? = when (inputStatus) {
+        Status.ACTIVE_IN.status -> true
+        Status.ACTIVE_OUT.status -> true
+        Status.INACTIVE_TRANSFER.status -> true
+        Status.INACTIVE_OUT.status -> false
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/prisoner/CurrentlyManaged.kt:20]: what is the difference between `false` and `null`?

**Review by markreesmoj — COMMENTED** [review]
Looks OK, but do we really need to distinguish between false and null if someone is not currently managed? It would be simpler if it was just true/false

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/court/commonplatform/Defendant.kt (hunk 1 above)
Comment (H-Iwanejko) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/court/commonplatform/Defendant.kt]: Moved back 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/prisoner/CurrentlyManaged.kt (hunk 2 above)
Comment (H-Iwanejko) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/prisoner/CurrentlyManaged.kt:20]: I believe as its only for prisoner reocrds we dont wont the rest of the records to hold any value as per the AC's on: [this](https://dsdmoj.atlassian.net/browse/CPR-440)

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/prisoner/CurrentlyManaged.kt:20]: I am questioning the ACs themselves. What information do we gain from `null`?

Also `currently managed`is a concept in probation as well as prison (naturally enough).
Are we ever going to store that?

Comment (H-Iwanejko) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/prisoner/CurrentlyManaged.kt:20]: Yeah there was a discussion around this. I dont recall most of it but we know for sure with a certain level for prison that they are within our care. But under probation is subjective. Probably worth asking Tony to clarify this.

**Review by markreesmoj — APPROVED** [review]
👍

**Pull request merged**